.github/workflows/appsignals-e2e-eks-test.yml

```diff
@@ -107,21 +91,102 @@ jobs:
         if: inputs.caller-workflow-name == 'main-build'
         run: "sed -i 's#image:.*#image: ${{ inputs.appsignals-adot-image-name }}#g' instrumentation.yaml"
 
-      - name: Enable App Signals
+      - name: Remove log group deletion command
+        if: always()
         run: |
-          ./enable-app-signals.sh \
-          ${{ inputs.test-cluster-name }} \
-          ${{ env.AWS_DEFAULT_REGION }} \
-          ${{ env.SAMPLE_APP_NAMESPACE }}
-
-      # Application pods need to be restarted for the
-      # app signals instrumentation to take effect
-      - name: Restart the app pods
-        run: kubectl delete pods --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+          delete_log_group="aws logs delete-log-group --log-group-name '${{ env.LOG_GROUP_NAME }}' --region \$REGION"
+          sed -i "s#$delete_log_group##g" clean-app-signals.sh
 
-      - name: Wait for sample app pods to come up
+      - name: Deploy sample app via terraform and wait for the endpoint to come online
+        id: deploy-sample-app
+        working-directory: testing/terraform/eks
         run: |
-          kubectl wait --for=condition=Ready pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }}
+          terraform init
+          terraform validate
+          
+          # Attempt to deploy the sample app on an EKS instance and wait for its endpoint to come online. 
+          # There may be occasional failures due to transitivity issues, so try up to 2 times. 
+          # Success of 0 indicates that both the terraform deployment and the endpoint are running, while 1 indicates
+          # that it failed at some point
+          retry_counter=0
+          max_retry=2
+          success=1 
+          while [ $success -eq 1 ]; do
+            if [ $retry_counter -ge $max_retry ]; then
+              echo "Max retry reached, failed to deploy terraform and connect to the endpoint. Exiting code"
+              exit 1
+            fi
+            echo "Attempt $retry_counter"
+            success=0
```

Comment:
Success should be made 0 *after* the `terraform apply` command has completed. That's is the assumption in the following code.

Reply:
Success of 1 indicates that the setting up App Signals on the sample app failed, while 0 indicates that everything ran successfully. 

The logic is:
1. Set Success to 1 (Set initial value to 1 so that the while loop runs)
2. While Success is 1 (Indicates that terraform deployment/endpoint connection failed and will try again):
     2a:  Set Success to 0 (Set the value to 0 and if there were any failures change it to 1)
     2b: Run Terraform apply (If the deployment failed, then success will change to 1)
     2c: If Success is still 0, then install App Signals and check endpoint connection
     2d: If endpoint connection failed, change success to 1
     2e: If Success is 1 at this point, then either the deployment or connection failed and run the while loop again.       If it is still 0, then the code ran successfully and exit the while loop

If the success is made 0 after the terraform apply, then it will override whether terraform deployment succeeded or not. If after the terraform deployment the success is 1, we want to skip the endpoint connection step and redeploy the terraform again.

Reply:
I think it's a little confusing because 1 usually means `true` and in this case setting `success` to 1 means the previous step failed, i.e. success is not true.

Also, could we not just wrap the retry in a while look that checks the retry counter instead of the success variable? Technically, success is a variable specific to a single attempt and shouldn't even be used outside the loop or as a condition for it. Instead, we should rely on the retry counter, i.e. `if attempt # < attempt limit, run retry`